.bit/responses/3.2-Week 3 Step 2.md

```diff
@@ -1,80 +1,262 @@
+
+![https://user-images.githubusercontent.com/69332964/99161780-8cf0c680-26c3-11eb-9bfc-78dc3262b038.png](https://user-images.githubusercontent.com/69332964/99161780-8cf0c680-26c3-11eb-9bfc-78dc3262b038.png)
+
+You will need to upgrade your storage account because Event Grid Subscriptions will only work with a v2 version. Follow this [tutorial](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-upgrade?tabs=azure-portal) to upgrade it.
+
+### Writing our *First* Azure Function to Upload an Image
+
+⬇ **Some housekeeping...**
+
+- For the function to work, we have to initialize the packages/libraries we installed in the beginning of part 1.
+- Take note of the `process.env` value being assigned to `connectionstring` in the code below (*Line 3*). Use this [tutorial](https://docs.microsoft.com/en-us/azure/azure-functions/functions-how-to-use-azure-function-app-settings) to add in your own secret strings from your storage container.
+    - The storage container is the one you created when you started your Function App. Navigate to it and find your secret strings here:
+
+    ![https://user-images.githubusercontent.com/69332964/99161798-ba3d7480-26c3-11eb-8e55-eac4bd4cb174.png](https://user-images.githubusercontent.com/69332964/99161798-ba3d7480-26c3-11eb-8e55-eac4bd4cb174.png)
+
+    ![https://user-images.githubusercontent.com/69332964/99161822-ec4ed680-26c3-11eb-8977-f12beb496c24.png](https://user-images.githubusercontent.com/69332964/99161822-ec4ed680-26c3-11eb-8977-f12beb496c24.png)
+
+    - Keep these safe, and use the connection string in the corresponding variable in the code.
+    - *Note: You'll need to store other strings in environment variables later on in the tutorial*
+
+Let's start just by initializing a few variables we'll need. In your `initializing.js` file: 
 
-<details>
-<summary>:exclamation: I'm stuck. Can I get some hints?</summary>
-  </br>
-  
-Fill in the gaps:
-  
 ```js
+var multipart = require("parse-multipart");
+const { BlobServiceClient } = require("@azure/storage-blob");
+const connectionstring = process.env["AZURE_STORAGE_CONNECTION_STRING"];
+const account = "<YOUR ACCOUNT NAME>";
+```
+
+⬇ **The main block of code**
+
+- Notice that we are able to name the file with the user's username in line 10 by receiving it from the header.
+    - Later on in the JS, we will send the username in the header of the request.
+- The `parse-multipart` library is being used in lines 4-11 to parse the image from the POST request we will later make with the frontend; refer to the documentation linked above.
+- Some if-else logic is used from lines 13-22 to determine the file extension.
+- We then call the `uploadBlob()` function in line 24.
 
+`uploadTrigger.js`:
+
+```js
 module.exports = async function (context, req) {
-    context.log(‘JavaScript HTTP trigger function processed a request.‘);

```

Comment:
Not all of this code will be used, check out these solutions for clarification https://github.com/bitprj/Intro-To-Serverless/pull/149/files#diff-6efcd774313f5a855464f808e148b39714c4d58f4e994aa73b091b8f99b8255f